File: app/src/main/java/com/scatl/uestcbbs/module/message/model/MessageModel.java
package com.scatl.uestcbbs.module.message.model;

import android.content.Context;

import com.alibaba.fastjson.JSONObject;
import com.scatl.uestcbbs.entity.AtMsgBean;
import com.scatl.uestcbbs.entity.DianPingMessageBean;
import com.scatl.uestcbbs.entity.PrivateChatBean;
import com.scatl.uestcbbs.entity.PrivateMsgBean;
import com.scatl.uestcbbs.entity.ReplyMeMsgBean;
import com.scatl.uestcbbs.entity.SendPrivateMsgResultBean;
import com.scatl.uestcbbs.entity.SimplePostListBean;
import com.scatl.uestcbbs.entity.SystemMsgBean;
import com.scatl.uestcbbs.entity.UploadResultBean;
import com.scatl.uestcbbs.helper.rxhelper.Observer;
import com.scatl.uestcbbs.util.ForumUtil;
import com.scatl.uestcbbs.util.RetrofitCookieUtil;
import com.scatl.uestcbbs.util.RetrofitUtil;
import com.scatl.uestcbbs.util.SharePrefUtil;

import java.io.File;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;
import okhttp3.MediaType;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import retrofit2.Call;


/**
 * author: sca_tl
 * description:
 * date: 2020/1/27 15:45
 */
public class MessageModel {
    public void getSystemMsg( int page,
                              int pageSize,
                              Observer<SystemMsgBean> observer) {
        Observable<SystemMsgBean> observable = RetrofitUtil
                .getInstance()
                .getApiService()
                .systemMsg(page, pageSize);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getAtMeMsg(int page,
                           int pageSize,
                           Observer<AtMsgBean> observer) {
        Observable<AtMsgBean> observable = RetrofitUtil
                .getInstance()
                .getApiService()
                .atMsg(page, pageSize);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getReplyMsg(int page,
                            int pageSize,
                            Observer<ReplyMeMsgBean> observer) {
        Observable<ReplyMeMsgBean> observable = RetrofitUtil
                .getInstance()
                .getApiService()
                .replyMeMsg(page, pageSize);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getPrivateMsg(String json,
                              Observer<PrivateMsgBean> observer) {
        Observable<PrivateMsgBean> observable = RetrofitUtil
                .getInstance()
                .getApiService()
                .privateMsg(ForumUtil.getAppHashValue(), json);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getPrivateChatMsgList(String json,
                                      Observer<PrivateChatBean> observer) {
        Observable<PrivateChatBean> observable = RetrofitUtil
                .getInstance()
                .getApiService()
                .privateChatMsgList(json);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void sendPrivateMsg(String json,
                               Observer<SendPrivateMsgResultBean> observer) {
        Observable<SendPrivateMsgResultBean> observable = RetrofitUtil
                .getInstance()
                .getApiService()
                .sendPrivateMsg(json);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getDianPingMsg(int page, Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getDianPingMsg1(page);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void deleteAllPrivateMsg(int uid, String formHash, Observer<String> observer) {
        Map<String, String> map = new HashMap<>();
        map.put("deletepm_deluid[]", uid + "");
        map.put("custompage", "1");
        map.put("deletepmsubmit_btn", "true");
        map.put("deletesubmit", "true");
        map.put("formhash", formHash);

        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .deletePrivateMsg(RetrofitCookieUtil.generateRequestBody(map));
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void deleteSinglePrivateMsg(int pmid, int touid, String formHash, Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .deleteSinglePrivateMsg(formHash, "pmdeletehk_" + pmid, touid, pmid);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void uploadImages(Context context,
                             List<File> files,
                             String module,
                             String type,
                             Observer<UploadResultBean> observer) {
        Map<String, RequestBody> params = new HashMap<>();

        params.put("module", RequestBody.create(MediaType.parse(""), module));
        params.put("type", RequestBody.create(MediaType.parse(""), type));
        params.put("accessToken", RequestBody.create(MediaType.parse(""), SharePrefUtil.getToken(context)));
        params.put("accessSecret", RequestBody.create(MediaType.parse(""), SharePrefUtil.getSecret(context)));

        List<MultipartBody.Part> parts = new ArrayList<>(files.size());
        for (File file : files) {
            RequestBody requestBody = RequestBody.create(MediaType.parse("image/jpeg"), file);
            MultipartBody.Part part = MultipartBody.Part.createFormData("uploadFile[]", file.getName(), requestBody);
            parts.add(part);
        }

        Observable<UploadResultBean> observable = RetrofitUtil
                .getInstance()
                .getApiService()
                .uploadImage(params, parts);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getUserSpace(int uid, String doo, Observer<String> observer) {
        Observable<String> observable = RetrofitUtil
                .getInstance()
                .getApiService()
                .userSpace(uid, doo);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/message/model/MessageModel.kt
package com.scatl.uestcbbs.module.message.model

import com.scatl.uestcbbs.entity.AtMsgBean
import com.scatl.uestcbbs.entity.DianPingMsgBean
import com.scatl.uestcbbs.entity.PrivateChatBean
import com.scatl.uestcbbs.entity.PrivateMsgBean
import com.scatl.uestcbbs.entity.ReplyMeMsgBean
import com.scatl.uestcbbs.entity.SendPrivateMsgResultBean
import com.scatl.uestcbbs.entity.SystemMsgBean
import com.scatl.uestcbbs.entity.UploadResultBean
import com.scatl.uestcbbs.http.BaseBBSResponseBean
import com.scatl.uestcbbs.util.ForumUtil
import com.scatl.uestcbbs.util.RetrofitCookieUtil
import com.scatl.uestcbbs.util.RetrofitUtil
import io.reactivex.Observable
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.MultipartBody
import okhttp3.RequestBody
import okhttp3.RequestBody.Companion.asRequestBody
import okhttp3.RequestBody.Companion.toRequestBody
import java.io.File

/**
 * Created by sca_tl at 2023/6/8 16:28
 */
class MessageModel {
    fun getSystemMsg(page: Int, pageSize: Int): Observable<SystemMsgBean> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .systemMsg(page, pageSize)
    }

    fun getAtMeMsg(page: Int, pageSize: Int): Observable<AtMsgBean> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .atMsg(page, pageSize)
    }

    fun getReplyMsg(page: Int, pageSize: Int): Observable<ReplyMeMsgBean> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .replyMeMsg(page, pageSize)
    }

    fun getPrivateMsg(json: String?): Observable<PrivateMsgBean> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .privateMsg(ForumUtil.getAppHashValue(), json)
    }

    fun getPrivateChatMsgList(json: String?): Observable<PrivateChatBean> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .privateChatMsgList(json)
    }

    fun sendPrivateMsg(json: String?): Observable<SendPrivateMsgResultBean> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .sendPrivateMsg(json)
    }

    fun deleteAllPrivateMsg(uid: Int, formHash: String): Observable<String> {
        val map: MutableMap<String, String> = HashMap()
        map["deletepm_deluid[]"] = uid.toString() + ""
        map["custompage"] = "1"
        map["deletepmsubmit_btn"] = "true"
        map["deletesubmit"] = "true"
        map["formhash"] = formHash

        return RetrofitCookieUtil
            .getInstance()
            .apiService
            .deletePrivateMsg(RetrofitCookieUtil.generateRequestBody(map))
    }

    fun deleteSinglePrivateMsg(pmid: Int, touid: Int, formHash: String?): Observable<String> {
        return RetrofitCookieUtil
            .getInstance()
            .apiService
            .deleteSinglePrivateMsg(formHash, "pmdeletehk_$pmid", touid, pmid)
    }

    fun uploadImages(files: List<File>, module: String, type: String): Observable<UploadResultBean> {
        val params: MutableMap<String, RequestBody> = HashMap()
        params["module"] = module.toRequestBody("".toMediaTypeOrNull())
        params["type"] = type.toRequestBody("".toMediaTypeOrNull())

        val parts: MutableList<MultipartBody.Part> = ArrayList(files.size)
        for (file in files) {
            val requestBody = file.asRequestBody("image/jpeg".toMediaTypeOrNull())
            val part = MultipartBody.Part.createFormData("uploadFile[]", "ImageFile.png", requestBody)
            parts.add(part)
        }

        return RetrofitUtil
            .getInstance()
            .apiService
            .uploadImage(params, parts)
    }

    fun getUserSpace(uid: Int, doo: String?): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .userSpace(uid, doo)
    }

    fun getDianPingMsg(page: Int, pageSize: Int): Observable<DianPingMsgBean> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .dianPingMsg(page, pageSize)
    }
}